eth2/utils/cached_tree_hash/src/vec_arena.rs

```diff
@@ -0,0 +1,435 @@
+use ssz::{Decode, Encode};
+use ssz_derive::{Decode, Encode};
+use std::marker::PhantomData;
+use std::ops::Range;
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum Error {
+    UnknownArenaId(usize),
+    OffsetOverflow,
+    OffsetUnderflow,
+    RangeOverFlow,
+}
+
+#[derive(Debug, PartialEq, Clone, Default, Encode, Decode)]
+pub struct VecArena<T: Encode + Decode> {
```

Comment:
Some comments in this module would be great, at the moment it's got me like :eyes: